altair/utils/core.py

```diff
@@ -602,6 +620,41 @@ def parse_shorthand(
     return attrs
 
 
+def infer_vegalite_type_for_dfi_column(
+    column: Column,
+) -> Union[_InferredVegaLiteType, Tuple[_InferredVegaLiteType, list]]:
+    from pyarrow.interchange.from_dataframe import column_to_array
+
+    try:
+        kind = column.dtype[0]
+    except NotImplementedError as e:
+        # Edge case hack:
+        # dtype access fails for pandas column with datetime64[ns] type,
```

Comment:
I'll take a look

Reply:
I opened one and added it to the comment in [5f4fc48](https://github.com/altair-viz/altair/pull/3114/commits/5f4fc487837c8535a9ca9dbe8b7a3c89f18d2ac0).

See https://github.com/pandas-dev/pandas/issues/54239

Reply:
Actually, this same error occurs when converting to arrow through the DataFrame interchange protocol with:

```python
from pyarrow.interchange import from_dataframe
from_dataframe(data)
```

Currently, we don't do this for pandas DataFrames because we use the pandas specific serialization logic. This is something we'll run into when trying to push pandas DataFrames through the DataFrame Interchange Protocol serialization logic.  I'm not sure of the best path forward here, but I don't think it's blocking for this PR.

What do you think?

Reply:
Not blocking here. Raising upstream and circumventing here is fine.
How to approach this best in the data serialization logic is another issue/PR👍